ffi/src/lib.rs

```diff
@@ -124,11 +222,17 @@ impl<T, E> From<Result<T, E>> for ExternResult where E: std::error::Error {
     }
 }
 
-// A store cannot be opened twice to the same location.
-// Once created, the reference to the store is held by the caller and not Rust,
-// therefore the caller is responsible for calling `destroy` to release the memory
-// used by the Store in order to avoid a memory leak.
-// TODO: Start returning `ExternResult`s rather than crashing on error.
+/// A store cannot be opened twice to the same location.
+/// Once created, the reference to the store is held by the caller and not Rust,
+/// therefore the caller is responsible for calling `destroy` to release the memory
+/// used by the Store in order to avoid a memory leak.
+/// TODO: Start returning `ExternResult`s rather than crashing on error.
```

Comment:
There's a pattern for this type of `Result` across the FFI boundary in mozilla-central, but the best reference I can find says that the situation is more complicated than I thought: https://groups.google.com/d/topic/mozilla.dev.platform/udd-DrIDYP4/discussion.

I think we should talk to a Rust/C expert to figure out a good pattern for this.  I doubt we want to proliferate `Extern*` types on the Rust FFI side of the house, but maybe I'm wrong.